accounts/safe7579/src/zkEmailRecovery/modules/SafeRecoveryModule.sol

```diff
@@ -75,26 +75,27 @@ contract SafeRecoveryModule is RecoveryModuleBase {
      * @param smartAccount The smart account to check
      * @return true if the module is initialized, false otherwise
      */
-    function isInitialized(
-        address smartAccount
-    ) external view override returns (bool) {
+    function isInitialized(address smartAccount) external view returns (bool) {
         return accountIsConfigured[smartAccount];
     }
 
     /*//////////////////////////////////////////////////////////////////////////
                                      MODULE LOGIC
     //////////////////////////////////////////////////////////////////////////*/
 
-    function recover(
+    function _recover(
         address account,
-        bytes[] memory subjectParams
-    ) external override {
-        if (msg.sender != zkEmailRecovery) {
-            revert NotTrustedRecoveryContract();
-        }
-
+        bytes[] calldata subjectParams
+    )
+        internal
+        override
+        onlyConfiguredAccount(account)
+    {
+        // prevent out of bounds error message, in case subject params are invalid
+        if (subjectParams.length < 3) revert InvalidSubjectParams();
         address oldOwner = abi.decode(subjectParams[1], (address));
         address newOwner = abi.decode(subjectParams[2], (address));
```

Comment:
Im not too familiar what you can inject here. would it be possible to inject previous owner from offchain in this array? if so, it would safe some gas.

if its not possible, maybe consider axing the isOwner call and implement that check in the getPreviousOwnerInLinkedList() function

Reply:
One option would be to pass it from the ZkEmailRecovery contract as one of the `subjectParams`. One issue with this is it would make the email subject extra crowded (it is already quite long) and that is meant to be a user facing message as well as a functional one. Having an extra previous owner could add confusion to the user 

The email subject for a safe is currently:
```
Recover account 0xE760ccaE42b4EA7a93A4CfA75BC649aaE1033095 from old owner 0x7c8999dC9a822c1f0Df42023113EDB4FDd543266 to new owner 0x7240b687730BE024bcfD084621f794C2e4F8408f using recovery module 0x6d2Fa6974Ef18eB6da842D3c7ab3150326feaEEC
```
With the previous owner it might be:
```
Recover account 0xE760ccaE42b4EA7a93A4CfA75BC649aaE1033095 from old owner 0x7c8999dC9a822c1f0Df42023113EDB4FDd543266 to new owner 0x7240b687730BE024bcfD084621f794C2e4F8408f with previous owner in linked list  0x7240b687730BE024bcfD084621f794C2e4F8408f using recovery module 0x6d2Fa6974Ef18eB6da842D3c7ab3150326feaEEC
```

Alternatively could pass it in another function call either on the recovery module or the zkEmailRecovery contract. This would be an extra step which harms ux slightly. Another thing to bear in mind is that a recovery config might stay in place for a long time (many years) before being used. So adding the previous owner to recovery config would run the risk of being invalid down the line if the owners linked list changes without going through recovery e.g. an account owner just rotating an owner normally. So these values would ideally be passed in the stage where the recovery request is generated (not recovery config which is setup when you install the recovery module). This then brings us back to the subject params as that would be the main avenue for adding additional arguments to this stage of the recovery flow.

Although this could still be a separate function call e.g. `addRecoveryParams`. We actually explored something like this and then removed it to improve UX. We had a `storeOwner` function on the recovery module https://github.com/rhinestonewtf/modulekit/blob/8537298d9341d880dfe8d727e0c755368bf6376b/accounts/safe7579/src/modules/SafeRecoveryModule.sol#L69-L76. This was a little awkward as this was when we were trying to use the same subject for Safe account recovery and a recovery for an OwnableValidator, so we had this approach of encoding the values in a create2 address. Won't go deep into why we were using a create2 address and not completely analogous to what we're talking about but to show as an example.

Let me know if you have any ideas here but the key goal for us is keeping UX as simple as we can, and if that is at the expense of gas efficiency then that is ok. 

> if its not possible, maybe consider axing the isOwner call and implement that check in the getPreviousOwnerInLinkedList() function
This makes sense will consider this!